sdk/core/azure-core-serializer-json-gson/src/test/java/com/azure/core/serializer/json/gson/GsonPropertyNameTests.java

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.serializer.json.gson;
+
+import com.google.gson.annotations.SerializedName;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import reactor.test.StepVerifier;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Member;
+import java.lang.reflect.Method;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class GsonPropertyNameTests {
+    private static final String expectValueInField = "expectFieldName";
+    private static GsonPropertyNameSerializer serializer;
+
+    @BeforeAll
+    public static void setup() {
+        serializer = new GsonPropertyNameSerializer();
+    }
+
+    @Test
+    public void testPropertyNameOnFieldName() throws NoSuchFieldException {
+        class Hotel {
+            String hotelName;
+        }
+        Field f = Hotel.class.getDeclaredField("hotelName");
+
+        assertMemberValue(f, "hotelName");
+    }
+
+    @Test
+    public void testPropertyNameOnFieldAnnotation() throws NoSuchFieldException {
+        class Hotel {
+            @SerializedName(value = expectValueInField)
+            String hotelName;
+        }
+        Field f = Hotel.class.getDeclaredField("hotelName");
+        assertMemberValue(f, expectValueInField);
+
+    }
+
+    @Test
+    public void testPropertyNameOnFieldAnnotationWithEmptyValue() throws NoSuchFieldException {
+        class Hotel {
+            @SerializedName(value = "")
+            String hotelName;
+        }
+        Field f = Hotel.class.getDeclaredField("hotelName");
+
+        assertMemberValue(f, "hotelName");
+    }
+
+    @Test
+    public void testPropertyNameOnMethodName() throws NoSuchMethodException {
+        class Hotel {
+            String hotelName;
+
+            public String getHotelName() {
+                return hotelName;
+            }
+        }
+
+        Method m = Hotel.class.getDeclaredMethod("getHotelName");
+
+        assertMemberValue(m, "getHotelName");
```

Comment:
Does GSON maintain full method names or does it drop Java bean notation? I know in Jackson's case this would serialize as `hotelName` (possibly `HotelName` as I'm not completely certain about how it handles casing).

Reply:
What you refer is the one without annotation on Method.
However, if we do not have annoatation, we cannot guarantee what we pass is the getter method with 'get' prefix. I don't think we need to check too much things here as validate if it is the invoked getter is a huge cost.